packages/grid/_modules_/grid/hooks/features/columns/useGridColumns.ts

```diff
@@ -108,7 +109,7 @@ const upsertColumnsState = (
   return newState;
 };
 
-export function useGridColumns(columns: GridColumns, apiRef: GridApiRef): void {
+export function useGridColumns(apiRef: GridApiRef, { columns }: { columns: GridColumns }): void {
```

Comment:
```suggestion
export function useGridColumns(apiRef: GridApiRef, { columns }: GridComponentProps): void {
```

Reply:
It's actually better to just put the props that you depend on instead of the larger object. As when the plugin api is properly done, then components might have different props types with more or less props so if we bind to the type it will create an error whereas if we pass only an object with columns then it will work. 
useGridColumns only care about columns so it doesn't need an object with filters.

Reply:
We can do `Pick<GridComponentProps, 'Columns'>`